### ISSUE TITLE
Specify indentation for esxml s-exps

### ISSUE THREAD

**Issue by aidalgol** [tali713/esxml#13]
It would be nice if esxml s-expressions were treated specially when indenting.  To start with, I would like to have

``` elisp
(tag ()
     body)
```

automatically indented as

``` elisp
(tag ()
  body)
```

I'm not sure how do accomplish this, but so far it looks as if we'll need something like `(put 'symbol 'lisp-indent-function INDENT-SPEC)`.

**Comment by alphapapa** [comment]
`(declare (indent defun))` is usually the solution.

**Comment by wasamasa** [comment]
Yeah, but that's for individual functions/macros.  What you'd need here is an indentation function that recognizes the context and adjusts indentation to be more DWIMish.  The proposed alternative of declaring indentation for each possible tag symbol isn't exactly actionable (just how many do you think exist for XML?) and will backfire if a user ever decides to call a function `html`.
